core/src/main/java/org/apache/calcite/rel/rules/SortRemoveConstantKeysRule.java

```diff
@@ -44,7 +45,7 @@
 
   private SortRemoveConstantKeysRule() {
     super(
-        operand(Sort.class, any()),
+        operand(LogicalSort.class, any()),
```

Comment:
Again this is a backwards incompatible change. Other than that I am more in favor of creating different instances of the rule rather than limiting the rule to match only LogicalSort. We have seen projects were people create their own set of logical operators. Changing the rule like this makes it impossible for such projects to use the rule.

Reply:
Exactly. The rule did work for the projects that inherit from Sort, however, it would stop working.

Reply:
Thanks for pointing this out. I am working on a revision.